third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -1087,20 +1087,25 @@ where
         _traversal_context: &mut TraversalContext,
         addr: AccountAddress,
         ty: &Type,
+        load_data: bool,
     ) -> PartialVMResult<DataCacheEntry> {
         let (entry, bytes_loaded) = TransactionDataCache::create_data_cache_entry(
             module_storage,
             resource_resolver,
             &addr,
             ty,
+            load_data,
         )?;
         gas_meter.charge_load_resource(
             addr,
             TypeWithRuntimeEnvironment {
                 ty,
                 runtime_environment: module_storage.runtime_environment(),
             },
-            entry.value().view(),
+            match entry.maybe_value() {
+                None => None,
+                Some(v) => v.view(),
+            },
             bytes_loaded,
```

Comment:
This means we double charge for
```
exists<A>
borrow<A>
```

We probably need to refactor this bit because if cache does not contain resource, it does not imply we have not charge for it (only first access is charged).